apps/public/src/app/(general)/bijlage/Bijlage.tsx

```diff
@@ -1,39 +1,66 @@
-            <Paragraph>
-              Voeg een foto toe om de situatie te verduidelijken. Verwijder alle persoonsgegevens van u en derden.
-            </Paragraph>
-            <UnorderedList>
-              <UnorderedList.Item>U kunt maximaal drie bestanden tegelijk toevoegen.</UnorderedList.Item>
-              <UnorderedList.Item>Toegestane bestandtypes: jpg, jpeg en png.</UnorderedList.Item>
-              <UnorderedList.Item>Een bestand mag maximaal 20 MB groot zijn.</UnorderedList.Item>
-            </UnorderedList>
-          </Column>
-          <FileUpload id="file-upload" aria-describedby="file-upload-description" />
-        </Field>
-        <SubmitButton>Volgende vraag</SubmitButton>
-      </form>
-    </Grid.Cell>
-  </Grid>
-)
+import { postAttachmentForm, redirectToNextPage } from './actions'
+
+const isErrorMessage = (obj: unknown): obj is { message: string } => {
+  return typeof obj === 'object' && obj !== null && 'message' in obj && typeof (obj as any).message === 'string'
+}
+
+export const Bijlage = () => {
+  const [uploadedFiles, setUploadedFiles] = useState<AttachmentOutput[]>([])
+  const [errorMessage, setErrorMessage] = useState<string>()
+
+  const handleOnChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (!event.currentTarget.files) return
+
+    const result = await postAttachmentForm(event.currentTarget.files)
+
+    if (isErrorMessage(result)) {
+      setErrorMessage(result.message)
+    } else {
+      result && setUploadedFiles((currentFiles) => [...currentFiles, ...result])
+    }
+  }
+
+  return (
+    <Grid paddingBottom="large" paddingTop="medium">
+      <Grid.Cell span={{ narrow: 4, medium: 6, wide: 6 }} start={{ narrow: 1, medium: 2, wide: 3 }}>
+        <BackLink href="/locatie">Vorige vraag</BackLink>
+        <Heading className="ams-mb--sm">Foto’s</Heading>
+        {errorMessage && <Paragraph>{errorMessage}</Paragraph>}
+        <form action={redirectToNextPage}>
+          <Field>
+            <Label htmlFor="file-upload" optional>
+              Heeft u een foto om toe te voegen?
+            </Label>
+            <Column id="file-upload-description" className="ams-mb--sm">
+              <Paragraph>
+                Voeg een foto toe om de situatie te verduidelijken. Verwijder alle persoonsgegevens van u en derden.
+              </Paragraph>
+              <UnorderedList>
+                <UnorderedList.Item>U kunt maximaal drie bestanden tegelijk toevoegen.</UnorderedList.Item>
+                <UnorderedList.Item>Toegestane bestandtypes: jpg, jpeg en png.</UnorderedList.Item>
+                <UnorderedList.Item>Een bestand mag maximaal 20 MB groot zijn.</UnorderedList.Item>
+              </UnorderedList>
+            </Column>
+            <FileUpload
+              id="file-upload"
+              handleOnChange={handleOnChange}
+              aria-describedby="file-upload-description"
+              uploadedFiles={uploadedFiles}
```

Comment:
I think it makes sense to separate the list from the input. The list is stored in `Bijlage`, it doesn't really need to be in `FileUpload`

Reply:
good idea! But it's not the scope of this ticket. [This ticket](https://gemeente-amsterdam.atlassian.net/jira/software/c/projects/SIG/boards/1243?selectedIssue=SIG-5986) is about building the list.